File: LinkedList/linkedlist.py
class Node:
    def __init__(self,value):
        self.value = value
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
    
    def append(self,value):
        new_node = Node(value)    
        if self.head is None:
            self.head = new_node
        else:
            current = self.head
            while current.next is not None:
                current = current.next
            current.next = new_node

    def pop_head(self):
        if not self.head:
            return None
        popped = self.head.value
        self.head = self.head.next
        return popped

linked_list = LinkedList()
linked_list.append(98)
linked_list.append(120)
linked_list.append(14)

print(linked_list.pop_head())
    

File: LinkedList/linkedlist.js
function Node(value)
{
    this.value = value;
    this.next = null;
}

function LinkedList()
{
    this.head = null;

    this.append = function(value)
        {
            let newNode = new Node(value);
            if(this.head == null)
            {
                this.head = newNode;
            }
            else
            {
                let current = this.head;
                while(current.next !== null)
                {
                    current = current.next;
                }
                current.next = newNode;
            }
        };

    this.popHead = function()
        {
            if(!this.head)
            {
                return null;
            }
            let popped = this.head.value;
            this.head = this.head.next;
            return popped;
        };
}

let list = new LinkedList();
list.append(98);
list.append(120);
list.append(14);

console.log(list.popHead());
    
